pkg/kubelet/client/kubelet_client.go

```diff
@@ -14,39 +14,61 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package unversioned
+package client
 
 import (
 	"errors"
+	"net"
 	"net/http"
+	"time"
 
+	"k8s.io/kubernetes/pkg/api"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
 	"k8s.io/kubernetes/pkg/util"
 )
 
+type KubeletClientConfig struct {
+	// Default port - used if no information about Kubelet port can be found in Node.NodeStatus.DaemonEndpoints.
+	Port        uint
+	EnableHttps bool
+
+	// TLSClientConfig contains settings to enable transport layer security
+	client.TLSClientConfig
+
+	// Server requires Bearer authentication
+	BearerToken string
+
+	// HTTPTimeout is used by the client to timeout http requests to Kubelet.
+	HTTPTimeout time.Duration
+
+	// Dial is a custom dialer used for the client
+	Dial func(net, addr string) (net.Conn, error)
+}
+
 // KubeletClient is an interface for all kubelet functionality
 type KubeletClient interface {
 	ConnectionInfoGetter
 }
 
 type ConnectionInfoGetter interface {
-	GetConnectionInfo(host string) (scheme string, port uint, transport http.RoundTripper, err error)
+	GetConnectionInfo(ctx api.Context, host string) (scheme string, port uint, transport http.RoundTripper, err error)
```

Comment:
sorry, meant s/host/nodeName/ throughout the whole PR for declarations/calls to this interface